backend/inspirehep/orcid/utils.py

```diff
@@ -226,3 +232,38 @@ def distributed_lock(lock_name, expire=10, auto_renewal=True, blocking=False):
 
 class DistributedLockError(Exception):
     pass
+
+
+def update_moved_orcid(old_orcid, new_orcid):
+    author_record = AuthorsRecord.get_record_by_pid_value(old_orcid, "orcid")
+    if new_orcid not in get_value(author_record, "ids.value", []):
+        new_author_ids = [
+            {"schema": "ORCID", "value": new_orcid},
+            *author_record["ids"],
+        ]
+        author_record["ids"] = new_author_ids
+        author_record.update(dict(author_record))
+        db.session.commit()
+    user_orcid_account = RemoteAccount.query.filter(
+        RemoteAccount.user_id == UserIdentity.id_user, UserIdentity.id == old_orcid
+    ).one()
+    remove_access_token_for_orcid_account(user_orcid_account, old_orcid, new_orcid)
+    LOGGER.info(
+        "ORCID updated",
+        new_orcid=new_orcid,
+        old_orcid=old_orcid,
+        orcid_account=user_orcid_account.id,
+    )
+
+
+def remove_access_token_for_orcid_account(user_orcid_account, orcid, new_orcid):
+    access_token = user_orcid_account.remote_tokens[0]
+    delete_access_token(access_token.access_token, orcid)
+    user_orcid_account.extra_data["orcid"] = new_orcid
+    user_orcid_account.extra_data["allow_push"] = False
```

Comment:
interesting that this works thanks to using the correct `Mutable` SQLAlchemy pattern, unlike what they do in `invenio-records` (@pazembrz will be please :smiley: ) https://github.com/inveniosoftware/invenio-oauthclient/blob/19f0e0c73e62ac1a85285612c754e290c4faa9fa/invenio_oauthclient/models.py#L53.